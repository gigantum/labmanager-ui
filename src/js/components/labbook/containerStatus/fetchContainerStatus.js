//vendor
import {
  graphql,
} from 'react-relay'
//environment
import {fetchQuery} from 'JS/createRelayEnvironment';

const containerStatusQuery = graphql`
  query fetchContainerStatusQuery($name: String!, $owner: String!, $first: Int!){
  labbook(name: $name, owner: $owner){
    environment{
      containerStatus
      imageStatus
    }
    activityRecords(first: $first){
      edges{
        node{
          id
        }
      }
    }
  }
}
`

const FetchContainerStatus = {
  getContainerStatus: (owner, labbookName) =>{
    const variables = {
      'owner': owner,
      'name': labbookName,
      'first': Math.floor(Math.random() * 10000)
      }

    return new Promise((resolve, reject) =>{

      let fetchData = function(){

        fetchQuery(containerStatusQuery(), variables).then((response) => {

          resolve(response.data)

        }).catch((error) =>{
          console.log(error)
          reject(error)
        })
      }

      fetchData()
    })
  }
}

export default FetchContainerStatus
