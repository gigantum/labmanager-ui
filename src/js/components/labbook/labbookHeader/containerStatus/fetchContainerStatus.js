// vendor
import { graphql } from 'react-relay';
// environment
import { fetchQuery } from 'JS/createRelayEnvironment';

const containerStatusQuery = graphql`
  query fetchContainerStatusQuery($name: String!, $owner: String!){
  labbook(name: $name, owner: $owner){
    environment{
      containerStatus
      imageStatus
    }
  }
}
`;

const FetchContainerStatus = {
  getContainerStatus: (owner, labbookName) => {
    const variables = {
      owner,
      name: labbookName,
    };

    return new Promise((resolve, reject) => {
      const fetchData = function () {
        fetchQuery(containerStatusQuery(), variables).then((response) => {
          resolve(response.data);
        }).catch((error) => {
          console.log(error);
          reject(error);
        });
      };

      fetchData();
    });
  },
};

export default FetchContainerStatus;
