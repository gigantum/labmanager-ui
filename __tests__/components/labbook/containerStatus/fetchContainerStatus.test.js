
      import React from 'react'
      import renderer from 'react-test-renderer';
      import {mount} from 'enzyme'
      import fetchContainerStatus from 'Components/labbook/containerStatus/fetchContainerStatus';

      import relayTestingUtils from 'relay-testing-utils'

      test('Test fetchContainerStatus', () => {

        const wrapper = renderer.create(

           <fetchContainerStatus />

        );

        const tree = wrapper.toJSON()

        expect(tree).toMatchSnapshot()

      })