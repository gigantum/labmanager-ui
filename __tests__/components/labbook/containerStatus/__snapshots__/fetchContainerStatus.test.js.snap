// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test fetchContainerStatus 1`] = `<fetchContainerStatus />`;
